python/ray/tune/suggest/variant_generator.py

```diff
@@ -1,19 +1,23 @@
 import copy
 import logging
 from collections.abc import Mapping
-from typing import Any, Dict, Generator, List, Optional, Tuple
+from typing import Any, Dict, Generator, List, Optional, Tuple, Union
 
 import numpy
 import random
 
 from ray.tune import TuneError
-from ray.tune.sample import Categorical, Domain, Function
+from ray.tune.sample import (Categorical, Domain, Function,
+                             _BackwardsCompatibleNumpyRng, np_random_generator)
 
 logger = logging.getLogger(__name__)
 
 
 def generate_variants(unresolved_spec: Dict,
-                      constant_grid_search: bool = False
+                      constant_grid_search: bool = False,
+                      random_state: Optional[Union[
+                          _BackwardsCompatibleNumpyRng, "np_random_generator",
+                          numpy.random.RandomState, int]] = None
```

Comment:
This should also import and use RandomState:

```suggestion
                      random_state: RandomState = None
```